src/basicProof/BasicProvers.sml

```diff
@@ -4,6 +4,16 @@
        is overkill.
  ---------------------------------------------------------------------------*)
 
+(* This file as a whole is assumed to be under the license in the file
+"COPYRIGHT" in the HOL4 distribution (note added by Mario Castelán         UOK
+Castro).
+
+For the avoidance of legal uncertainty, I (Mario Castelán Castro) hereby   UOK
+place my modifications to this file in the public domain per the Creative
+Commons CC0 1.0 public domain dedication <https://creativecommons.org/publ
+icdomain/zero/1.0/legalcode>. This should not be interpreted as a personal
+endorsement of permissive (non-Copyleft) licenses. *)
+
 structure BasicProvers :> BasicProvers =
```

Comment:
Such copyright notices are not used normally in HOL. I know that  @marioxcc believes they are necessary and tries to push people towards using them. I believe otherwise. If someone adds them to their own files, thats their own business. But before adding them to something central like BasicProvers, some general discussion on how to do it for HOL is needed, I believe. If we do it, it should be done in the same fashion for all files.

Reply:
Thanks for creating the separate issue #559. Let's resolve this there, ok?

Reply:
Yes, let's resolve it there. However, having this copyright notice in is a blocking issue for me at the moment.